### PR TITLE
Insertion from the media library to the canvas

### PR DESCRIPTION
**Pull request by wassgha**
ezgif com-video-to-gif (23)

### Changes
- Fixes issue with drop targets for elements coming from the media library panel
- Implements insertion of elements from the media panel to the canvas
- Adds animations to drag/drop from the media panel
- Implements correct sizing and placement of elements when inserted

### CODE REVIEW

File: assets/src/edit-story/components/canvas/canvasDropzone.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+ */
+import StoryPropTypes from '../../types';
+import { useDropTargets } from '../dropTargets';
+import { useUnits } from '../../units';
+import useCanvas from './useCanvas';
+import useInsertElement from './useInsertElement';
+
+const Container = styled.div``;
+
+function CanvasDropzone({ children }) {
```
Comment (dvoytenko) [inline assets/src/edit-story/components/canvas/canvasDropzone.js]: Let's call it `CanvasElementDropzone` - that'd be less ambiguous.

File: assets/src/edit-story/components/canvas/canvasDropzone.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  const {
+    state: { pageContainer },
+  } = useCanvas();
+  const {
+    actions: { editorToDataX, editorToDataY },
+  } = useUnits();
+
+  const onDropHandler = useCallback(
+    (e) => {
+      const isMedia = e.dataTransfer.types.includes('resource/media');
```
Comment (dvoytenko) [inline assets/src/edit-story/components/canvas/canvasDropzone.js]: I'm hearing that `.includes()` is problematic here. See https://github.com/google/web-stories-wp/blob/master/assets/src/edit-story/components/uploadDropTarget/getDragType.js. If true - let's refactor somehow?

File: assets/src/edit-story/components/canvas/canvasDropzone.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    [
+      activeDropTargetId,
+      pageContainer,
+      insertElement,
+      editorToDataX,
+      editorToDataY,
+    ]
+  );
+  const onDragOverHandler = (e) => {
+    e.stopPropagation();
```
Comment (dvoytenko) [inline assets/src/edit-story/components/canvas/canvasDropzone.js]: Only cancel on `e.dataTransfer.types.includes('resource/media')`.

File: assets/src/edit-story/components/canvas/canvasLayout.js [hunk 4]
```diff
@@ -46,12 +48,16 @@ function CanvasLayout() {
 
   return (
     <Background ref={backgroundRef} style={layoutParamsCss}>
-      <SelectionCanvas>
-        <DisplayLayer />
-        <NavLayer />
-        <FramesLayer />
-      </SelectionCanvas>
-      <EditLayer />
+      <CanvasUploadDropTarget>
```
Comment (dvoytenko) [inline assets/src/edit-story/components/canvas/canvasLayout.js:51]: Bad rebase?

File: assets/src/edit-story/components/canvas/canvasDropzone.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+/**
+ * Internal dependencies
+ */
+import StoryPropTypes from '../../types';
+import { useDropTargets } from '../dropTargets';
+import { useUnits } from '../../units';
+import useCanvas from './useCanvas';
+import useInsertElement from './useInsertElement';
+
+const Container = styled.div``;
```
Comment (dvoytenko) [inline assets/src/edit-story/components/canvas/canvasDropzone.js]: Let's also do `width: 100%; height: 100%` here for consistency.

**Review by dvoytenko — CHANGES_REQUESTED** [review]

File: assets/src/edit-story/components/canvas/canvasLayout.js (hunk 4 above)
Comment (wassgha) [reply assets/src/edit-story/components/canvas/canvasLayout.js:51]: ?

File: assets/src/edit-story/components/canvas/canvasDropzone.js (hunk 2 above)
Comment (wassgha) [reply assets/src/edit-story/components/canvas/canvasDropzone.js]: Addressed

File: assets/src/edit-story/components/canvas/canvasLayout.js (hunk 4 above)
Comment (dvoytenko) [reply assets/src/edit-story/components/canvas/canvasLayout.js:51]: hmm. fine now.

**Review by dvoytenko — APPROVED** [review]

**Pull request merged**